### PR TITLE
implementation of tower of hanoi game algorithm

### PR DESCRIPTION
**Pull request by willx75**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [ ] **If creating a new file :**
  - [ ] added links to it in the README files ?
  - [ ] included tests with it ?
  - [ ] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

- [ ] **if done some changes :**
  - [ ] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

### CODE REVIEW

File: algorithms/games/hanoi.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+mouvement = 0
+
+
+def TowerOfHanoi(n, source, dest, by):
```
Comment (keon) [inline algorithms/games/hanoi.py]: needs to be underscore

File: algorithms/games/hanoi.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+mouvement = 0
+
+
+def TowerOfHanoi(n, source, dest, by):
+    global mouvement
```
Comment (keon) [inline algorithms/games/hanoi.py]: global variable isn't allowed

File: algorithms/games/hanoi.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+mouvement = 0
+
+
+def TowerOfHanoi(n, source, dest, by):
+    global mouvement
+    mouvement += 1
+    if n == 1:
+        print('Move the plate from ', source, ' to ', dest)
```
Comment (keon) [inline algorithms/games/hanoi.py]: no print statement (you could make it an optional flag for printing logs)

File: algorithms/games/hanoi.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    if n == 1:
+        print('Move the plate from ', source, ' to ', dest)
+    else:
+        TowerOfHanoi(n-1, source, by, dest)
+        print('Move the plate from ', source, ' to ', dest)
+        TowerOfHanoi(n-1, by, dest, source)
+
+
+# Driver Code
+n = int(input("Enter the number of plates : "))
```
Comment (keon) [inline algorithms/games/hanoi.py]: needs to be unittested

Comment (willx75) [reply algorithms/games/hanoi.py]: should I test if it's an integer ?

**Comment by keon** [comment]
Hi @willx75 thanks for addressing the feedback, but the coding style still does not conform with the the rest of the code. (such as Tower_Of_Hanoi -> tower_of_hanoi).
the variable name should also be in english :)

File: algorithms/games/hanoi.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# @author willx75
+# Tower of Hanoi recursion game algorithm is a game, it consists of three rods and a number of disks of different sizes, which can slide onto any rod
+
+import logging
+
+log = logging.getLogger()
+logging.basicConfig(level=logging.DEBUG)
+
+
+def Tower_Of_Hanoi(n, source, dest, by, mouvement):
```
Comment (ankit167) [inline algorithms/games/hanoi.py]: It would be better to add comments, what each parameter means

**Comment by keon** [comment]
Closing as stale — the repo was completely restructured in v1.0.0. Thank you for the contribution\!
